snapcraft/plugins/go.py

```diff
@@ -67,18 +58,40 @@ def env(self, root):
                 '$LDFLAGS'
             ]).format(root, snapcraft.common.get_arch_triplet()) + '"',
         ]
+        return env
 
     def pull(self):
         # use -d to only download (build will happen later)
         # use -t to also get the test-deps
-        self._run(['go', 'get', '-t', '-d', self.fullname])
+        super().pull()
```

Comment:
On Fri, Nov 13, 2015 at 9:30 PM, Leo Arias <notifications@github.com> wrote:

> In snapcraft/plugins/go.py
> <https://github.com/ubuntu-core/snapcraft/pull/97#discussion_r44849654>:
>
> >
> >      def pull(self):
> >          # use -d to only download (build will happen later)
> >          # use -t to also get the test-deps
> > -        self._run(['go', 'get', '-t', '-d', self.fullname])
> > +        super().pull()
>
> So if the source is git://name/project, will this get the repo two times?
> first pull will download it to src, and then _local_pull will download it
> to src/go/src?
> I'm not sure if I'm missing something.
>
super().pull() will bring in the source (git://name/project)

self._local_pull() brings in the dependencies for that source.